src/operator/tensor/square_sum-inl.h

```diff
@@ -28,13 +28,16 @@ inline bool SquareSumForwardInferStorageType(const nnvm::NodeAttrs& attrs,
                                              std::vector<int>* out_attrs) {
   CHECK_EQ(in_attrs->size(), 1U);
   CHECK_EQ(out_attrs->size(), 1U);
-  CHECK_EQ((*in_attrs)[0], kRowSparseStorage)
-    << "_square_sum only supports row-sparse ndarray as input";
   const ReduceAxesParam& param = nnvm::get<ReduceAxesParam>(attrs.parsed);
-  if (param.axis[0] == 1 && param.keepdims) {  // sum per row and keep dims
-    STORAGE_TYPE_ASSIGN_CHECK(*out_attrs, 0, kRowSparseStorage);
-  } else {
-    STORAGE_TYPE_ASSIGN_CHECK(*out_attrs, 0, kDefaultStorage);
+  if (in_attrs->at(0) == kRowSparseStorage) {  // current impl
+    if (param.axis[0] == 1 && param.keepdims) {  // sum per row and keep dims
+      STORAGE_TYPE_ASSIGN_CHECK(*out_attrs, 0, kRowSparseStorage);
+    } else {
+      STORAGE_TYPE_ASSIGN_CHECK(*out_attrs, 0, kDefaultStorage);
+    }
+  } else {  // fallback
+      STORAGE_TYPE_ASSIGN_CHECK(*in_attrs, 0, kDefaultStorage);
+      STORAGE_TYPE_ASSIGN_CHECK(*out_attrs, 0, kDefaultStorage);
```

Comment:
In this case, since there's no FCompute operator for square_sum, we have nothing to fallback on. In general, if we have such FCompute available, it's not appropriate to `STORAGE_TYPE_ASSIGN_CHECK` directly since one of the input/output could already be assigned with KRowSparse/KCSRStorage. Using `STORAGE_TYPE_ASSIGN_CHECK` will halt the program; instead we should add fallback in FComputeEx so that if in_stype = undefined, out_stype = csr, we are still able to fallback and call FCompute. What do you think?